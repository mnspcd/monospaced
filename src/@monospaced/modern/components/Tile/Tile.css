.mdrn .Tile {
  align-items: center;
  border: calc(1rem / 16) solid var(--color-grey-2);
  color: var(--color-blue-9);
  display: flex;
  font-family: var(--font-monospaced);
  font-variant: slashed-zero;
  justify-content: center;
  padding: 1.5em 0.75em;
  width: 100%;
}

.mdrn .Tile--onDarkBackground {
  border-color: var(--color-grey-8);
}
