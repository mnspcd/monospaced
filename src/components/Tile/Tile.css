.Tile {
  align-items: center;
  border: calc(1rem / 16) solid var(--mnspcd-color-blue-2);
  color: var(--mnspcd-color-blue-9);
  display: flex;
  font-family: var(--font-monospaced);
  font-variant: slashed-zero;
  justify-content: center;
  padding: 1.5em 0.75em;
  width: 100%;
}

.Tile--onDarkBackground {
  border-color: var(--mnspcd-color-blue-8);
}
