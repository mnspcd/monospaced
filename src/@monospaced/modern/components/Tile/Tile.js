import classNames from "classnames";
import PropTypes from "prop-types";
import React, { useContext } from "react";

import { SurfaceContext } from "../Surface";

import "./Tile.css";

/**
 * Use `Tile` to display content in a block link.
 */
const Tile = ({ children, href, label, width }) => {
  const surfaceBackground = useContext(SurfaceContext);

  return (
    <a
      aria-label={label}
      className={classNames({
        Tile: true,
        [`Tile--onDarkBackground`]: surfaceBackground === "dark",
      })}
      href={href}
      style={{ width }}
    >
      {children}
    </a>
  );
};

Tile.propTypes = {
  /**
   * Content to display in the Tile, usually an SVG component.
   */
  children: PropTypes.node.isRequired,

  /**
   * HTML href attribute of the Tile link.
   */
  href: PropTypes.string.isRequired,

  /**
   * Accessible label text for the Tile link.
   */
  label: PropTypes.string,

  /**
   * Valid CSS width value for the Tile.
   */
  width: PropTypes.string,
};

export default Tile;
