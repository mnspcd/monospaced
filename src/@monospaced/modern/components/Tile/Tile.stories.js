import React from "react";
import { select, text } from "@storybook/addon-knobs";

import Group from "../Group";
import Logomark from "../Logomark";
import Surface from "../Surface";

import Tile from "./Tile";

export default {
  component: Tile,
  title: "Components/Tile",
};

export const basic = () => (
  <Tile href="#" label="Monospaced" width="8.5em">
    <Logomark width="78.72%" />
  </Tile>
);

export const onDarkBackground = () => (
  <Surface backgroundColor="grey-dark">
    <div style={{ padding: "1em" }}>
      <Tile href="#" label="Monospaced" width="8.5em">
        <Logomark width="78.72%" />
      </Tile>
    </div>
  </Surface>
);

export const tessellation = () => (
  <Group tessellate>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
    <Group.Item>
      <Tile href="https://example.com" label="Label">
        <Logomark width="100%" />
      </Tile>
    </Group.Item>
  </Group>
);

export const knobs = () => (
  <Surface
    backgroundColor={select(
      "backgroundColor",
      Surface.backgroundColors,
      "white",
      "Surface",
    )}
  >
    <Tile
      href={text("href", "https://example.com")}
      label={text("label", "Label")}
      width={text("width", "8.5em")}
    >
      <Logomark width="78.72%" />
    </Tile>
  </Surface>
);

knobs.story = { parameters: { docs: { disable: true } } };
