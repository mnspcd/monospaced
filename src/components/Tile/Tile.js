import PropTypes from "prop-types";
import React from "react";
import classNames from "classnames";

import { SurfaceConsumer } from "../Surface";
import "./Tile.css";

const Tile = ({ children, href, width }) => (
  <SurfaceConsumer>
    {surfaceBackground => (
      <a
        className={classNames({
          Tile: true,
          [`Tile--onDarkBackground`]: surfaceBackground === "dark",
        })}
        href={href}
        style={{ width }}
      >
        {children}
      </a>
    )}
  </SurfaceConsumer>
);

Tile.propTypes = {
  children: PropTypes.node.isRequired,
  href: PropTypes.string,
  width: PropTypes.string,
};

export default Tile;
