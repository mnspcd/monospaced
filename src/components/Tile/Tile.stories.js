import React from "react";
import { select, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Group from "../Group/Group";

import Tile from "./Tile";

const icons = require("../../assets/clients");

storiesOf("Components/Tile", module)
  .add("Component", () => {
    const href = text("href", "https://example.com");
    const Icon = icons[select("image", Object.keys(icons), "BladudFlies")];

    return (
      <Tile href={href} width={text("width", "8.5em")}>
        <Icon width="100%" />
      </Tile>
    );
  })
  .add("Tessellation", () => {
    return (
      <Group tessellate>
        {Object.keys(icons).map(key => {
          const Icon = icons[key];

          return (
            <Group.Item key={key}>
              <Tile href={text("href", "https://example.com")}>
                <Icon width="100%" />
              </Tile>
            </Group.Item>
          );
        })}
      </Group>
    );
  });
