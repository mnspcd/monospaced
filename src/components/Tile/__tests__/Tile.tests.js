/* eslint-env jest */
import React from "react";
import { render } from "enzyme";

import Tile from "../Tile";

describe("Tile component", () => {
  it("should render correctly", () => {
    const component = render(<Tile>children</Tile>);
    expect(component).toMatchSnapshot();
  });
});
