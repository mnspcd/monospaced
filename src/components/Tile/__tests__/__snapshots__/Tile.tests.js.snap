// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile component should render correctly 1`] = `
<a
  class="Tile"
>
  children
</a>
`;
